src/CraftingRecipes.h

```diff
@@ -170,8 +179,14 @@ class cCraftingRecipes
 
 	/** Searches for anything dye related for leather, calculates the appropriate color value, and sets the resulting value. */
 	void HandleDyedLeather(const cItem * a_CraftingGrid, cCraftingRecipes::cRecipe * a_Recipe, int a_GridStride, int a_GridWidth, int a_GridHeight);
-} ;
-
 
+private:
+	/** Checks if all ingredients of the recipe are within the know items list and if the (new) ItemId is part of the ingredients. This makes sure to only find 'new' recipes */
+	bool isNewCraftableRecipe(const cRecipe * a_Recipe, const cItem & a_Item, const std::set<cItem, cItem::sItemCompare> & a_KnownItems);
 
+	/** Mapping the minecraft recipe names to the internal cuberite recipe Ids */
+	std::map<AString, UInt32> m_RecipeNameMap;
```

Comment:
Please put field declarations above function declarations (within the `private` section of the class)